### ISSUE TITLE
Overlapping file types between slack and slackevents

### ISSUE THREAD

**Issue by abustany** [slack-go/slack#1053]
What's the rationale behind having two slightly different file types, one in the root package and one in `slackevents`? Would a PR to merge them have a chance to be accepted?

**Comment by ArcticSnowman** [comment]
I agree there are several types that are identical and could be combined into a single definition.
I trying to add SocketMode support to an app that works with  RTM and I'm having to write little convertor function to map `slackevent.Something` to `slack.Something` rather than just having a single definition.

**Comment by nlopes** [comment]
Closing as a duplicate. This is tracked in #1546, which is the successor to #723 for unifying types
  between the slack and slackevents packages. This will be part of a breaking-change batch.
